packages/fxa-auth-server/test/local/payments/paypal-processor.js

```diff
@@ -0,0 +1,351 @@
+      mockStripeHelper.updateInvoiceWithPaypalTransactionId = sandbox.fake.resolves(
+        {}
+      );
+      mockStripeHelper.payInvoiceOutOfBand = sandbox.fake.resolves({});
+      const result = await processor.handlePaidTransaction(unpaidInvoice, [
+        { status: 'Completed', transactionId: 'test1234' },
+      ]);
+      assert.isTrue(result);
+      sinon.assert.calledOnceWithExactly(
+        mockStripeHelper.updateInvoiceWithPaypalTransactionId,
+        unpaidInvoice,
+        'test1234'
+      );
+    });
+
+    it('returns true and logs if > 1 success', async () => {
+      mockStripeHelper.updateInvoiceWithPaypalTransactionId = sandbox.fake.resolves(
+        {}
+      );
+      mockStripeHelper.payInvoiceOutOfBand = sandbox.fake.resolves({});
+      mockLog.error = sandbox.fake.returns({});
+      const result = await processor.handlePaidTransaction(unpaidInvoice, [
+        { status: 'Completed', transactionId: 'test1234' },
+        { status: 'Completed', transactionId: 'test12345' },
+      ]);
+      assert.isTrue(result);
+      sinon.assert.calledOnceWithExactly(
+        mockStripeHelper.updateInvoiceWithPaypalTransactionId,
+        unpaidInvoice,
+        'test1234'
+      );
+      sinon.assert.calledOnceWithExactly(
+        mockLog.error,
+        'multipleCompletedTransactions',
+        {
+          customer: unpaidInvoice.customer,
+          invoiceId: unpaidInvoice.id,
+          transactionCount: 2,
+          excessTransactions: ['test12345'],
+        }
+      );
+    });
+  });
+
+  describe('handlePendingTransaction', () => {
+    it('returns true if a pending within grace period exists', async () => {
+      processor.inGracePeriod = sandbox.fake.returns(true);
+      const result = await processor.handlePendingTransaction(unpaidInvoice, [
+        { status: 'Pending' },
+      ]);
+      assert.isTrue(result);
+      sinon.assert.calledOnceWithExactly(
+        processor.inGracePeriod,
+        unpaidInvoice
+      );
+    });
+
+    it('returns true and logs if multiple pending within grace exist', async () => {
+      processor.inGracePeriod = sandbox.fake.returns(true);
+      mockLog.error = sandbox.fake.returns({});
+      const result = await processor.handlePendingTransaction(unpaidInvoice, [
+        { status: 'Pending' },
+        { status: 'Pending' },
+      ]);
+      assert.isTrue(result);
+      sinon.assert.calledOnceWithExactly(
+        processor.inGracePeriod,
+        unpaidInvoice
+      );
+      sinon.assert.calledOnceWithExactly(
+        mockLog.error,
+        'multiplePendingTransactions',
+        { customer: unpaidInvoice.customer, invoiceId: unpaidInvoice.id }
+      );
+    });
+
+    it('returns false if no pending exist', async () => {
+      processor.inGracePeriod = sandbox.fake.returns(true);
+      const result = await processor.handlePendingTransaction(unpaidInvoice, [
+        { status: 'Completed' },
+      ]);
+      assert.isFalse(result);
```

Comment:
I think the `false` returned here is from the 'Completed' status, not the grace period check.

Reply:
What do you mean?

Reply:
I think I misread the title on this one last night.